tempesta_fw/http_parser.c

```diff
@@ -59,6 +63,31 @@ __field_finish(TfwHttpMsg *hm, TfwStr *field,
 }
 
 /**
+ * Check whether a character is CR or LF.
+ */
+#define IS_CR_OR_LF(c) (c == '\r' || c == '\n')
+
+/**
+ * Scans the initial @n bytes of the memory area pointed to by @s for the first
+ * occurance of EOL character.
+ */
+static inline unsigned char *
+memchreol(const unsigned char *s, size_t n)
+{
+	while (n) {
+		if (IS_CR_OR_LF(*s))
+			return (unsigned char *)s;
+		s++, n--;
+	}
+	return NULL;
+}
```

Comment:
If ```strcspn()``` is just like ```memchreol()```, then why do we need to introduce our own function? One can find SSE implementation of ```strcspn()``` and can replace the poor kernel implementation by a better implementation of the standard functions. Meantime, it isn't obvious that ```memchreol()``` has many good SSE implementations.

I agree that it's better to do SSE optimizations in different patch. Just please make a TODO comment in the code or update #182 issue, such that we won't forget about the necessary optimizations.